lib/app/views/main_page.dart

```diff
@@ -52,26 +52,40 @@ class MainPage extends ConsumerWidget {
     );
 
     if (isAndroid) {
-      isNfcEnabled().then((value) =>
-          ref.read(androidNfcStateProvider.notifier).setNfcEnabled(value));
+      isNfcEnabled().then(
+          (value) => ref.read(androidNfcAdapterState.notifier).enable(value));
     }
 
     // If the current device changes, we need to pop any open dialogs.
-    ref.listen<AsyncValue<YubiKeyData>>(currentDeviceDataProvider, (_, __) {
-      Navigator.of(context).popUntil((route) {
-        return route.isFirst ||
-            [
-              'device_picker',
-              'settings',
-              'about',
-              'licenses',
-              'user_interaction_prompt',
-              'oath_add_account',
-              'oath_icon_pack_dialog',
-              'android_qr_scanner_view',
-              'android_alert_dialog'
-            ].contains(route.settings.name);
-      });
+    ref.listen<AsyncValue<YubiKeyData>>(currentDeviceDataProvider,
+        (prev, next) {
+      var canPop = true;
+      if ((next.value != null) && (prev?.value != null)) {
+        // if there is change only in fipsApproved, don't pop anything
+        var nextInfo = next.value!.info;
+        var prevInfo = prev!.value!.info;
+
+        canPop =
+            prevInfo.copyWith(fipsApproved: nextInfo.fipsApproved) != nextInfo;
+      } else if (next.hasValue && (prev != null && prev.isLoading)) {
+        canPop = false;
+      }
```

Comment:
This is intended to pop when the device changes, so a better check would be to avoid popping when serial doesn't change:
```
final serial = next.value?.info.serial;
if (serial != null && serial == prev.value?.info.serial) {
  return;
}

Navigator.of(context).popUntil((route) {
  ...
```

Reply:
thanks! this will avoid even more popping :)